### PR TITLE
Dockerize tests

### PR DESCRIPTION
**Pull request by lubien**
Implements #55 

Currently I'm pushing the image to Docker Hub as lubien/project_euler until @ryukinix makes a DestructHub account there.

Known issues:

- [ ] Need a DestructHub account on Docker Hub
- [ ] Bash solutions not work

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+version: "2"
+
+services:
+  runner:
+    image: lubien/project_euler
```
Comment (rafaelcn) [inline docker-compose.yml]: Until @ryukinix creates an account we may use your image, is that ok?

Comment (lubien) [reply docker-compose.yml]: As estated in the PR, that's the plan.
I've already asked him to do it but there where some issues on the website.

Comment (ryukinix) [reply docker-compose.yml]: Yes, i'm in trouble to create an account on docker hub because some seems fucked on forms to sign up.

**Comment by ryukinix** [comment]
Bash solution needs `gawk` installed on Docker image. This can be easily done, but is painful to push the image since each push is 1GB of upload. I wish if something like differential update was possible... :/

**Comment by lubien** [comment]
@rafaelcn Bash problem seems like a missing package gawk.

image

**Comment by lubien** [comment]
@ryukinix It used to be 1GB on one of my tests. After correctly doing stuff seems like it's about only half a gigabyte now.

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+                              --s python
+                              --s commonlisp
+                              --s ruby
+                              --s haskell
+                              --s go
+                              --s php
+                              --s python
+                              --s elixir
+                              --s lua
+                              --s c++
```
Comment (ryukinix) [inline docker-compose.yml]: @lubien you can enable `C` tests for while just to testing if works now?

Comment (lubien) [reply docker-compose.yml]: I know you wanna fuck me there and didn't even took me to a date but ok

**Comment by ryukinix** [comment]
Another issue:
- [ ] Need setup `locale` of Docker image to `utf-8` instead `latin1` to avoid warning on elixir executions

**Comment by lubien** [comment]
@ryukinix remember also about these python warning over the spinner.

**Comment by ryukinix** [comment]
Which warning?

**Comment by ryukinix** [comment]
This was not fixed?

**Comment by lubien** [comment]
@ryukinix https://travis-ci.org/lubien/ProjectEuler

At the end

**Comment by ryukinix** [comment]
ok

**Comment by ryukinix** [comment]
I need to updated to get gawk working. I'll install on #117 

Almost all the issues related on this PR has been fixed on #117.

**Pull request merged**